src/components/VisualQueryEditor.tsx

```diff
@@ -69,12 +70,20 @@ export const VisualQueryEditor: React.FC<Props> = props => {
   }, [datasourceId, databaseName, tableName]);
 
   const onAutoComplete = useCallback(
-    async (searchTerm?: string, column?: string) => {
-      const autoComplete = new AdxAutoComplete(datasource, tableSchema.value, databaseName, tableName);
-      const values = await autoComplete.search(searchTerm, column);
+    async (index: string, search: QueryEditorOperatorExpression) => {
+      const values = await datasource.autoCompleteQuery(
+        {
+          search,
+          database: databaseName,
+          expression: query.expression,
+          index,
+        },
+        tableSchema.value
+      );
+
       return values.map(value => ({ value, label: value }));
     },
-    [datasource, databaseName, tableName, tableSchema.value]
+    [datasource, databaseName, tableSchema.value, query.expression]
```

Comment:
nit: You should extract autoCompleteQuery from datasource and add it as a dependency.
```suggestion
    [autoCompleteQuery, databaseName, tableSchema.value, query.expression]
```

Reply:
Great catch!